src/app/features/settings/sign-out/sign-out.tsx

```diff
@@ -19,13 +20,18 @@ export function SignOutDialog({ isShowing, onUserDeleteWallet, onClose }: SignOu
       confirmBackup: whenWallet({ ledger: true, software: false }),
       confirmPasswordDisable: whenWallet({ ledger: true, software: false }),
     },
-    onSubmit() {
-      onUserDeleteWallet();
-    },
+    onSubmit() {},
   });
 
   const canSignOut = form.values.confirmBackup && form.values.confirmPasswordDisable;
 
+  function onClickSignOut() {
+    if (canSignOut) {
+      onClose();
+      onUserDeleteWallet();
+    }
+  }
```

Comment:
As this is the function handler, think we can have a better non-event based name, like `handleSignOut` or `signUserOutWhenPossible`